### PR TITLE
BPS Display PCB Revision

### PR DESCRIPTION
**Pull request by hayrenguul**
# Quality Assurance Checklist
To make reviews more efficient, please make sure the board meets the following standards and check everything off once the board meets the quality check. Once everything has been checked, the assigned reviewers will begin the review process. _Edit this description to check off the list._

There are exceptions with all guidelines. As long as your decisions are justified, then you are good! Contact the reviewers or the leads about any exceptions.

### Minimum Prerequisites
- [X] Pushed changes to branch.
- [X] The board passes ERC and DRC.
    - Note some ERC warnings are acceptable.
- [X] All traces are routed.
- [X] Units are in metric.

### 2D Spacing
- [X] The components are spaced out at an optimal distance.
    - All components can be easily hand-soldered.
    - IPC-SM-782A Standard requires a minimum distance of 1.0mm from edge to edge.
- [X] Components that are in parallel with each other are spaced out at an equal distance when possible.
- [x] The components are aligned with each other when possible.
- [x] Components are grouped based off of functionality.
    - E.g. all components for CAN should be grouped.
- [X] Bypass capacitors are less than 1cm away from their respective IC's power pins.

### 3D Spacing
- [x] Layout of components have been placed with mounting location and usage of the board in mind.
    - Are PCBs going to be stacked on above/below this board? Are tall components placed accordingly?
    - Are buttons and lights easily accessible and viewable?
    - When mounted into the car, are the heights of the components and connectors accounted for?
- [x] Location of connectors and wires going out of the board will prevent messy wiring.
    - Are all the wires that are going in the same direction placed on the same edge of the board?

### Components
- [x] Custom footprints have been double checked with the datasheet.
- [x] Pin 1 of the footprint is labeled in some way.

### Copper Layer
- [X] The trace widths and trace clearances are greater than JLCPCB's minimum requirements.
- [x] The trace lengths are as short as possible.
    - Can there be a more optimal route if you go to another layer?
- [x] Each trace's width is capable of handling the expected current flow.
    - Use PCB width calculator to calculate trace width.
- [x] *No sharp corners. No trace angles equal to or less than 90 degrees.
- [x] Traces are in parallel with each other when possible.
    - E.g. traces connected between an IC and MCU can be placed in parallel with each other.
- [x] There are no random trace appendages.
- [X] Vias placed in copper pads are fully encompassed in the copper pads.
- [X] Through-hole components do not have extraneous vias.

*Not really a problem for modern manufacturing techniques but good practice and important for high speed signal integrity.

### Silkscreen Layer
- [X] Visible text sizes are greater than JLCPCB's minimum requirements.
- [X] All visible text is on the silkscreen layer.
- [X] All reference labels of each component are not overlapping a copper pad or another component.
- [x] All connector pins are labeled with a meaningful and helpful name.
- [x] All LEDs are labeled with a meaningful and helpful name.

### Edge Cut Layer
- [X] The dimensions of the board and the mounting holes are nice values in metric i.e. no long decimals.
- [X] The physical outline of the board is on the edge cut layer.
- [X] Edge cuts are straight and parallel with opposite edge of the board.
- [X] Mounting holes are aligned.
- [X] Corners are curved and contoured to mounting holes.

### IMPORTANT NOTICE
- [X] I am confident that this board will be safe to use in a safety critical environment.
- [X] I had fun :)

## Other Comments
_Write any comments about the board that would help the reviewers here._

### CODE REVIEW

**Comment by SijWoo** [comment]
Heyo, I designed the original version of this board, but there were a lot of problems with my design. Here are some considerations I forgot to mention and some notes about the version's layout:
1. This was not accounted for in the original version of the display board but there needs to be 2 CAN connectors similar to how the CAN Sniffer Board has two.
2. This display board will be mounted on a movable arm. The mount will probably be connected south of the board. It would be ideal if all the wires come out of one edge of the board. The SWD programming pins can stay at the same place.
3. To make the traces out of the microcontroller more readable, the logic analyzer header can be separated to individual headers based off of the module. E.g. CAN TX, CAN RX, GND can be a 3 pin header close to the CAN transceiver IC. You can reference the Telemetry board, Controls Leader, or the BPS leader board for examples of the logic analyzer headers being broken out to separate headers. With this logic analyzer separation, the traces of the microcontroller will need to be retraced so trace following will be easier. Having all the logic analyzer points separate was not considered in the original display board.
4. This is the [display ](https://www.amazon.com/2-7inch-Resolution-Two-Color-Interface-Raspberry/dp/B07PKSZ3XK/ref=sr_1_9?crid=12ZSYAHHVYY3S&dchild=1&keywords=eink+display+waveshare&qid=1599930812&sprefix=eink+display+%2Caps%2C223&sr=8-9) that will mount on top of this board like an Arduino Shield. All connectors should be right angled so the two boards can stack on top of each other. As a note, the dimensions of this board is the exact same as the display board so any tall components on the top side of this board will get in the way. (11mm is the clearance between the two boards).
5. In the original version of the board, the PWR LED was placed on the right edge of the board. The board will be encased in a 3D printed case. The original plan was to have a small slit in the case so light can shine through and we can tell if the board is on or not. Placing the PWR LED to a location where the user can see it with the case on will be ideal.

**Comment by HannahSLee14** [comment]
Thanks for making the comment Sijin and thanks to both of you for putting in that work! I unchecked some of the items from the checklist so make sure to look back at the checklist to see what I unchecked and to fix those!

Some comments:
1. The 2x20 connector should be changed back to its original footprint. This board is going to have a shield (the eink display) which is going to be placed on top of this board. Having a horizontal connector defeats that purpose.
2. The logic analyzer pins can be left as vertical headers sticking out of the back. They are going to be used to test the board traces and connections so they won't actually be connected to the connector. Also, some of the placement will need to be changed (ie U2 is an actual component that has a height so putting a horizontal header over it will overlap and we will either not be able to put U2 on the board or not put J6 on the board - you will not need to worry about this anymore since you're changing the logic analyzer pins' footprints but it's good to make sure to consider the board in 3D). (Sijin, correct me if I'm wrong here and they actually do need to be horizontal.) 
3. The logic analyzer pins will need more descriptive names, particularly J6 and J9. One of them is for UART and the other is for CAN so labeling the pins the same way will lead to confusion. 
4. The CAN headers are still vertical. (Take a look back at Sijin's comment).
5. The traces from the microcontroller are going to be really hard to follow and will make debugging very difficult. You don't have to keep the same microcontroller placement as the original board, so feel free to move things around to optimize the organization and neatness of the traces. 
6. Grouping the components can be improved. The logic analyzer pins are a bit scattered and can be placed closer to their corresponding pins on the microcontroller. This will also help with comment 5. 
7. This may not be an issue, so I will double-check this and get back to both of you. But there's a chance having all the wires coming out of the south part of the board will interfere with the mounting of the board if the board is going to be mounted to the south. I'm not 100% sure this will be an issue or not, but I just wanted to let both of you know just in case. 

Thank you both for your hard work! 

EDIT: Ignore comment 7. The connector placement on the south edge looks good!

**Comment by HannahSLee14** [comment]
It looks really good!! You cleaned up a lot of the traces, and it's almost ready. Just a few small things:

1. There's a random trace appendage on the ground pad of C5 - double-check to make sure that there aren't any others. I know it's easy to miss these when you are rerouting traces.
2. Move the connectors on the bottom so that their bottom edges are on the edge cut line.
3. Rotate JP1 by 90 degrees clockwise. Since a jumper is going to be connected there, we want to make sure that nothing is in the way. 
4. Redo the board name to match the naming convention of the other boards. It will look something like:
   UTSVT - BPS
   BPS Display v2.0
   Fall 2020
   (You can also add a fun note or reference underneath a component!)

Thank you - you're doing great!

**Comment by HannahSLee14** [comment]
Just a couple more things!
1. Can you make sure to update the filled zone and run the design rules checker one more time? The filled zones haven't updated since when you moved the components around. You can do this by pressing b on the keyboard or by running the design rules checker. 
2. This is just a personal preference but can you make sure that T4 and T5 are aligned with each other and T4 and T6 as well? 
3. On the CAN IC (U2), I am a little concerned with how close the trace connecting pads 11 and 15 are to pad 12 (even though it passes the DRC). I would have that trace go on the back side of the board instead next to the CAN_H trace. 

image

4. Similar concern for the 3.3V trace connecting 3.3V of U2 to C19 since that's really close to GND and if it accidentally connects, it would short. I would just move the 3.3V trace to be more in the middle of U2 and C19. 

image

The board looks good - nice job!

**Comment by HannahSLee14** [comment]
@ClarkPoon @chaseblock @SijWoo Is it okay if one of you also reviews the board as well? I want to make sure that the board fits what BPS is looking for as well.

**Comment by SijWoo** [comment]
With the power LED at the bottom right corner, this is not important to be visible but not sure how the mount will look. I think the LED should be moved close to the right side just in case.

Besides that, I think the board looks pretty good

**Comment by SijWoo** [comment]
Just realized, would be good to have the test points for +5V, GND, +12V, PWRGND, CAN_L, CAN_H have silkcreen on the back of the board. Also adding a +3.3V testpoint would be useful
